### PR TITLE
Bump Matter Server to 0.1.0

### PR DESCRIPTION
**Pull request by balloob**
https://github.com/home-assistant-libs/python-matter-server/releases/tag/0.1.0

### CODE REVIEW

File: matter_server/Dockerfile [hunk 1]
```diff
@@ -109,13 +107,13 @@ COPY --from=builder-chip \
...
-     /usr/src/python-matter-server/python_matter_server-0.0.1-py3-none-any.whl \
+     /usr/src/python-matter-server/python_matter_server-${MATTER_SERVER_VERSION}-py3-none-any.whl \
      /tmp
 
-# hadolint ignore=DL3013
+# hadolint ignore=DL3013,SC2086
 RUN \
     pip3 install --no-cache-dir /tmp/chip-0.0-cp37-abi3-linux_*.whl \
-    && pip3 install --no-cache-dir /tmp/python_matter_server-0.0.1-py3-none-any.whl \
+    && pip3 install --no-cache-dir /tmp/python_matter_server-${MATTER_SERVER_VERSION}-py3-none-any.whl \
```
Comment (balloob) [inline matter_server/Dockerfile:116]: This doesn't work and somehow we end up with `/tmp/python_matter_server--py3-none-any.whl`

File: matter_server/Dockerfile [hunk 2]
```diff
@@ -109,13 +107,13 @@ COPY --from=builder-chip \
      /usr/src/connectedhomeip/out/python_lib/controller/python/chip-0.0-cp37-abi3-linux_*.whl \
      /tmp
 COPY --from=builder-matter-server \
-     /usr/src/python-matter-server/python_matter_server-0.0.1-py3-none-any.whl \
+     /usr/src/python-matter-server/python_matter_server-${MATTER_SERVER_VERSION}-py3-none-any.whl \
```
Comment (balloob) [inline matter_server/Dockerfile:110]: This doesn't work and somehow we end up with `/tmp/python_matter_server--py3-none-any.whl`

File: matter_server/Dockerfile [hunk 3]
```diff
@@ -64,7 +64,7 @@ FROM builder-base AS builder-matter-server
 
 WORKDIR /usr/src
 
-ARG MATTER_SERVER_VERSION=ac5545bcdfdf02a583ae44ebc0b2ab4c4e3686c5
+ARG MATTER_SERVER_VERSION=0.1.0
```
Comment (pvizeli) [inline matter_server/Dockerfile:67]: move this into build.json

Comment (pvizeli) [reply matter_server/Dockerfile:67]: https://github.com/home-assistant/addons/blob/master/ssh/build.yaml#L12

Comment (frenck) [reply matter_server/Dockerfile]: Done 👍

**Pull request merged**